Added mentions and company models

derekjou opened the pull request:
Modified settings to refer to company model, which was newly added. The company document will reference mention documents.

derekjou committed:
Added mentions and company models

Modified settings to refer to company model, which was newly added. The company document will reference mention documents.

ncen5293 commented:
When I make a new account, it makes a user document then crashes the backend and is unable to make a user settings document.
You would also have to change how we get and set companies 
![image](https://user-images.githubusercontent.com/28994838/69925370-cf52d080-147d-11ea-9400-e81c1e5981c2.png)

derekjou committed:
new model integration

Used references to settings on registration and created the company model on registration as well. Added findOrCreate to make process easier. Defaulted platform object in settings to clean up registration route.

derekjou commented:
> When I make a new account, it makes a user document then crashes the backend and is unable to make a user settings document.
> You would also have to change how we get and set companies
> ![image](https://user-images.githubusercontent.com/28994838/69925370-cf52d080-147d-11ea-9400-e81c1e5981c2.png)

Fixed! I won't be able to integrate the mentions model fully yet until we get the OK from Aecio on the interface. Reddit and Twitter routes will have to be modified for this as well later!

derekjou committed:
wrong feature

these routes are part of different feature so removing

derekjou committed:
Added link

Added link to the post into the model

derekjou committed:
Updated settings route

Fixed all routes to account for model change

ncen5293 reviewed (commented):
Everything looks good except you could just merge dev into this branch and in the /:email route just get the companies the same way you did with /:email/company for using the interface and giving the company names to the frontend

derekjou committed:
Merge branch 'dev' into mentions-model

derekjou committed:
Modified interface argument

Changed argument to reflect new model

ncen5293 commented on server/routes/settings.js:
```diff
@@ -48,23 +58,28 @@ router.put("/:email/platform/:platform", jwtVerify, (req, res, next) => {
+      settings.companies.forEach(company => companyNames.push(company.name));
       const interface = new Interface();
-      const mentions = await interface.getNewestMentions(settings.companies);
+      const mentions = await interface.getNewestMentions(companyNames);
```
can you also make it an async route since it is using await?
and can you pass the companyNames to the frontend too

derekjou committed:
added missing async

fixed async/await error
